### PR TITLE
#24 Updating HV project pages

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: _config/site.yml [hunk 1]
```diff
@@ -111,16 +114,16 @@ projects:
         - name: Documentation
           href: '/validator/documentation'
           css_class: 'icon-fixed-width icon-book'
+        - name: FAQ
```
Comment (hferentschik) [inline _config/site.yml]: It was an active choice of having FAQ and the contributing guide as external links. One argument is about duplication of information and the other is that in particular the FAQ is much better of being a wiki. At least until we had another discussion about this I would like to keep the external links. If we make these pages part of the site, the wiki pages should be deleted.

**Comment by hferentschik** [comment]
I think it would be best to split this pull request up into the part which fixes the announcement urls, typos and other things which have changed since I created the first HV skeleton pages and the part which pulls in FAQ and contribute. I am not convinced yet that this is the best approach.

**Comment by gunnarmorling** [comment]
Sure, we can split it; I'll update the PR. But I think it makes sense to have this information ultimately here:
- For Search we also have contribution guide and FAQ here, and consistency between projects would be nice
- The new web site finally gives us the chance to have all relevant information in one place, easily editable and nicely looking. What's the benefit of splitting up information between several resources/domains?

Of course the info would be removed from the wiki once the new web site is live, so I'm not concerned about duplication.

**Comment by emmanuelbernard** [comment]
The reason I internalized FAQ and contribution pages for Search is that it will increase search engine ranking for that piece of data and that in practice these pages are not edited by the community at large - or at least we tend to vet changes. Additional benefit is a more consistent look and feel.
FYI I'm also contemplating including the latest stable version of the documentation on the website itself but that has more challenges.

**Comment by hferentschik** [comment]
> For Search we also have contribution guide and FAQ here, and consistency between projects would be nice

Right, but I never said I liked that Search does it this way. And consistency is good, but not a definitive must IMO.

> The new web site finally gives us the chance to have all relevant information in one place, easily editable and nicely looking. What's the benefit of splitting up information between several resources/domains?

Because a website is not a wiki. I think we agreed that we still need a wiki, but that we are not going to make any tool changes there for now. And yes, awestruct make things easier, but involves still more steps than a wiki edit and a potential short downtime of the site during redeploy. Also people can not easily comment or even contribute if the FAQ and contribute page are part of the website. 

At the very least this needs to be discussed more. Think also about the different goes on how to build ORM version xyz. They are also on the wiki.

**Comment by hferentschik** [comment]
> The reason I internalized FAQ and contribution pages for Search is that it will increase search engine ranking for that piece of data 

How so?

> and that in practice these pages are not edited by the community at large - or at least we tend to vet changes. 

But sometimes people comment on it and to make small changes it is nice to be able to do them there and then in the browser.

> Additional benefit is a more consistent look and feel.

That's for sure an argument I understand. My point is just that there are reasons for these pages to be wiki pages as well and we should not just ignore them.

> FYI I'm also contemplating including the latest stable version of the documentation on the website itself but that has more challenges.

Right, there are many things to consider in this case. Not sure whether it is worth the effort but for sure an interesting idea.

**Comment by gunnarmorling** [comment]
> > For Search we also have contribution guide and FAQ here, and consistency between projects would be nice
> 
> Right, but I never said I liked that Search does it this way. And consistency is good, but not a definitive must IMO.

Agreed in general, but I think we should be able to align on this kind of thing to create a professional experience for our users.

> > The new web site finally gives us the chance to have all relevant information in one place, easily editable and nicely looking. What's the benefit of splitting up information between several resources/domains?
> 
> Because a website is not a wiki. I think we agreed that we still need a wiki, but that we are not going to make any tool changes there for now. And yes, awestruct make things easier, but involves still more steps than a wiki edit and a potential short downtime of the site during redeploy. 

I didn't experience any issues with that so far neither with beanvalidation.org nor my own awestruct site. It's really just copying HTML files.

> Also people can not easily comment or even contribute if the FAQ and contribute page are part of the website. 

But how often does that happen really? For the HV FAQ and contribution pages, there is exactly zero edits by community members and one single comment from 2011. The pages surely are user editable also with awestruct and if we like, we also can add commenting to these pages.

I'm not arguing against a wiki in general, only these pages don't seem to be the best candidates IMO. Others, more user-driven contents might make more sense in the Wiki (e.g. the "Frameworks integrating with HV" page).

> At the very least this needs to be discussed more.

Ok.

**Comment by gunnarmorling** [comment]
Re-ordered the commits; this branch/PR now only contains the non FAQ/contribution changes. The latter are backed up at https://github.com/gunnarmorling/hibernate.org/tree/HV_with_FAQ_and_contribution_pages.

**Comment by hferentschik** [comment]
Thanks :-)

**Comment by emmanuelbernard** [comment]
> And consistency is good, but not a definitive must IMO.

I will quote you on that in the future :D

**Comment by emmanuelbernard** [comment]
> Because a website is not a wiki. I think we agreed that we still need a wiki, but that we are not going to make any tool changes there for now. And yes, awestruct make things easier, but involves still more steps than a wiki edit and a potential short downtime of the site during redeploy. Also people can not easily comment or even contribute if the FAQ and contribute page are part of the website.

If you check arquillian.org, they have some pages with an `Edit this file` button that directly goes to the edit mode of GitHub or ask you to fork if you have not.
Here is an example of such a page http://arquillian.org/guides/getting_started/

There still is some friction to edit or propose a change but it's only one / two steps more than a wiki and everything is done on the browser.

**Comment by emmanuelbernard** [comment]
To be fair, on a personal note I like wikis less than I used to. The disadvantages I see in them might not be true in absolute but are generally. Likewise they might not apply to our specific case:
- skinning is usually not good / done
- sites using a wiki tend to look less professional and more like "we had to have a web site"
- a lot of side features are polluting the true focus of a page / section

In the original hibernate.org site we used the wiki as CMS + Wiki, highly skinned it, and created team sanctioned sections only editable to team members and free edit wikis for community contribution. I do like this split between team sanctioned vs other community contribution as it clarify the quality / endorsement of the data at bay.

**Comment by hferentschik** [comment]
> >    And consistency is good, but not a definitive must IMO.
> 
> will quote you on that in the future :D

I was afraid that I would into my own foot with this one :-)

**Comment by hferentschik** [comment]
> If you check arquillian.org, they have some pages with an Edit this file button that directly goes to the edit mode of GitHub or ask you to fork if you have not.
> Here is an example of such a page http://arquillian.org/guides/getting_started/

Nice, I like ;-) This would indeed alleviate my concerns. Can we get this as well?

**Pull request closed without merging**